Mathlib/CategoryTheory/CatCommSq.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2023 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import Mathlib.CategoryTheory.Equivalence
+
+/-! 2-commutative squares of functors
+
+Similarly as `CommSq.lean` defines the notion of commutative squares,
+this file introduces the notion of 2-commutative squares of functors.
+
+If `T : C₁ ⥤ C₂`, `L : C₁ ⥤ C₃`, `R : C₂ ⥤ C₄`, `B : C₃ ⥤ C₄` are functors,
+then `[CatCommSq T L R B]` contains the datum of an isomorphism `T ⋙ R ≅ L ⋙ B`.
+
```

Comment:
Would you mind adding a "Future work" section here to explain where you're planning to go with this development?

Reply:
Thanks! I have just added this.